src/EndpointConfig.js

```diff
@@ -1,17 +1,100 @@
+import localforage from 'localforage';
+import {parse as urlParse, format as urlFormat} from 'url';
+
+const urlMap = {
+  local: `http://${window.location.hostname}:8899`,
+  'testnet-edge': 'http://edge.testnet.solana.com:8899',
+  'testnet-beta': 'http://beta.testnet.solana.com:8899',
+  testnet: 'http://testnet.solana.com:8899',
+  /*
+  TODO: Switch to TLS endpoints...
+  'testnet-edge': 'https://api.edge.testnet.solana.com',
+  'testnet-beta': 'https://api.beta.testnet.solana.com',
+  testnet: 'https://api.testnet.solana.com',
+  */
+};
+
+let endpointName = process.env.NODE_ENV === 'development' ? 'local' : 'testnet';
+
+export async function load() {
+  try {
+    const newEndpointName = await localforage.getItem('endpointName');
+    if (typeof urlMap[newEndpointName] === 'string') {
+      endpointName = newEndpointName;
+    }
+  } catch (err) {
+    console.log(
+      `Unable to load endpointName from localforage, using default of ${endpointName}: ${err}`,
+    );
+  }
+  console.log('EndpointConfig loaded. endpointName:', endpointName);
+}
+
+export function getEndpointName() {
+  return endpointName;
+}
+
+export function getEndpoints() {
+  return Object.keys(urlMap);
+}
+
+export function setEndpointName(newEndpointName) {
+  if (typeof urlMap[newEndpointName] !== 'string') {
+    throw new Error(`Unknown endpoint: ${newEndpointName}`);
+  }
+  endpointName = newEndpointName;
+  console.log('endpointName is now', endpointName);
+  localforage.setItem('endpointName', endpointName).catch(err => {
+    console.log(`Failed to set endpointName in localforage: ${err}`);
+  });
+}
+
+export function getRpcUrl() {
+  return urlMap[endpointName];
+}
+
+export function getApiUrl() {
+  const urlParts = urlParse(getRpcUrl());
+  urlParts.host = '';
+  if (urlParts.protocol === 'https:') {
+    urlParts.port = '3443';
+  } else {
+    urlParts.port = '3001';
+  }
+  const url = urlFormat(urlParts);
+  console.info('getApiUrl:', url);
```

Comment:
@mvines wondering if maybe we should do the console.log at a higher level to display the whole config as an object, rather than in the individual calls?

Reply:
Sure go for it